### PR TITLE
Changes to improve compatibility with agent and retrieval skill in python_sdk

### PR DESCRIPTION
**Pull request by dusvyat**
## Description

Fixing agent + retrieval skill after decoupling agent from langchain_handler https://github.com/mindsdb/mindsdb/pull/9481

https://github.com/mindsdb/mindsdb_python_sdk/pull/133 depends on this PR

Part of ML-103

## Type of change

(Please delete options that are not relevant)

- [x] 🐛 Bug fix (non-breaking change which fixes an issue)
- [ ] ⚡ New feature (non-breaking change which adds functionality)
- [ ] 📢 Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] 📄 This change requires a documentation update

## Verification Process

To ensure the changes are working as expected:

 - [x]   Test Location: tested using both text2sql and retrieval examples in python_sdk
 - [ ]   Verification Steps: Outline the steps or queries needed to validate the change. Include any data, configurations, or actions required to reproduce or see the new functionality.

## Additional Media:

- [ ] I have attached a brief loom video or screenshots showcasing the new functionality or change.

## Checklist:

- [ ] My code follows the style guidelines(PEP 8) of MindsDB.
- [ ] I have appropriately commented on my code, especially in complex areas.
- [ ] Necessary documentation updates are either made or tracked in issues.
- [ ] Relevant unit and integration tests are updated or added.

### CODE REVIEW

File: mindsdb/interfaces/storage/db.py [hunk 1]
```diff
@@ -462,6 +462,7 @@ class Agents(Base):
     project_id = Column(Integer, nullable=False)
 
     model_name = Column(String, nullable=True)
+    provider = Column(String, nullable=True)
```
Comment (ea-rus) [inline mindsdb/interfaces/storage/db.py:465]: if we move `provider` from params we have to create migration for:
- adding this column to database
- move values from params to new column for existed agents

Comment (dusvyat) [reply mindsdb/interfaces/storage/db.py:465]: Thanks! I added the migrations to git, forgot to add them before

File: mindsdb/migrations/versions/2024-07-18_459a4cd24933_merge_divergent_branches.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+Revises: cbedc4968d5d, bfc6f44f5bc9
+Create Date: 2024-07-18 23:38:29.998117
+
+"""
+import mindsdb.interfaces.storage.db  # noqa
+
+
+# revision identifiers, used by Alembic.
+revision = '459a4cd24933'
+down_revision = ('cbedc4968d5d', 'bfc6f44f5bc9')
```
Comment (ea-rus) [inline mindsdb/migrations/versions/2024-07-18_459a4cd24933_merge_divergent_branches.py:13]: I think it should be fixed by https://github.com/mindsdb/mindsdb/pull/9518/files

File: mindsdb/migrations/versions/2024-07-19_45eb2eb61f70_add_provider_to_agent.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    with op.batch_alter_table('agents', schema=None) as batch_op:
+        batch_op.add_column(sa.Column('provider', sa.String(), nullable=True))
+
+    with op.batch_alter_table('chat_bots', schema=None) as batch_op:
```
Comment (ea-rus) [inline mindsdb/migrations/versions/2024-07-19_45eb2eb61f70_add_provider_to_agent.py:26]: db.py doesn't have changes about chat_bots table

File: mindsdb/migrations/versions/2024-07-18_459a4cd24933_merge_divergent_branches.py (hunk 2 above)
Comment (dusvyat) [reply mindsdb/migrations/versions/2024-07-18_459a4cd24933_merge_divergent_branches.py:13]: Ok will remove

File: mindsdb/migrations/versions/2024-07-19_45eb2eb61f70_add_provider_to_agent.py (hunk 3 above)
Comment (dusvyat) [reply mindsdb/migrations/versions/2024-07-19_45eb2eb61f70_add_provider_to_agent.py:26]: This is from auto generated changes, let me remove

**Review by ea-rus — APPROVED** [review]

**Pull request merged**